app/grandchallenge/algorithms/views.py

```diff
@@ -495,10 +496,25 @@ def create_upload(image_files):
             ci = interfaces[slug]
             if ci.is_image_kind:
                 if value:
-                    # create civ without image, image will be added when import completes
-                    civ = ComponentInterfaceValue.objects.create(interface=ci)
-                    civs.append(civ)
-                    upload_pks[civ.pk] = create_upload(value)
+                    widget = form.data[f"WidgetChoice-{ci.slug}"]
+                    if widget == WidgetChoices.IMAGE_SEARCH:
+                        (
+                            civ,
+                            created,
+                        ) = ComponentInterfaceValue.objects.get_or_create(
+                            interface=ci, image=value
+                        )
+                        if created:
+                            civ.full_clean()
+                            civ.save()
+                        civs.append(civ)
+                    elif widget == WidgetChoices.IMAGE_UPLOAD:
+                        # create civ without image, image will be added when import completes
+                        civ = ComponentInterfaceValue.objects.create(
+                            interface=ci
+                        )
+                        civs.append(civ)
+                        upload_pks[civ.pk] = create_upload(value)
```

Comment:
We probably should have an `else` block here that raises a runtime exception to remind future developers to extend the block when changes are made, you never know.